QUARKUS-2812.md

```diff
@@ -0,0 +1,51 @@
+# QUARKUS-2812 Build time metrics for Quarkus
+
+JIRA link: https://issues.redhat.com/browse/QUARKUS-2812.
+
+The aim of this feature is to collect build-time data in a Quarkus application and send them to an external data platform
+for analytical purposes.
+
+## Scope of testing
+
+What data is collected in what event.
+- Events:
+  - Dev mode build - should send a minimal data payload (a "ping").
+  - Production build (JVM, native) - should send full analytics data.
+- Data:
+  - Build type.
+  - Quarkus version, Java vendor and version, GraalVM/Mandrel version, OS name and version, build tool and its version (Gradle, Maven).
+  - Anonymous User ID, anonymous Project ID.
+  - List of extensions and their versions.
+  - Build duration, build errors
+
+Verify the correct destination of the data.
+- How to test that the correct data has been collected. Mock the destination, query it?
```

Comment:
Is there a schema definition for the sent data?

Reply:
No formal schema is available ATM. Only an example in the charter document.